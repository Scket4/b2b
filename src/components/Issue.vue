<template>
	<div class="issue">
		<div class="issue__main-info">
			<div class="issue__tag" @click="openIssueInfo">
				{{ issue.title }}
			</div>
			<div class="issue__author">
				<p class="issue__author--date">
					Автор: {{ issue.user.login }} от {{ dateCreated }}
				</p>
			</div>
		</div>
		<div class="issue__number"># {{ issue.number }}</div>
	</div>
</template>

<script>
export default {
	name: 'Issue',
	props: {
		issue: {
			type: Object,
			default: () => {
				return {};
			}
		}
	},
	computed: {
		dateCreated() {
			return new Date(this.issue.created_at).toLocaleDateString();
		}
	},
	methods: {
		openIssueInfo() {
			this.$router.push({
				name: 'issue',
				params: {
					id: this.issue.number,
					issue: JSON.stringify(this.issue)
				}
			});
		}
	}
};
</script>

<style lang="sass" solid>
@import '../variables/variables.sass'

.issue
	width: 100%
	border-bottom: 1px solid $grey
	padding: 10px 20px
	display: flex
	justify-content: space-between
	align-items: center
	&__main-info
		display: flex
		flex-direction: column
		justify-content: space-between
		align-items: flex-start
	&__tag
		font-weight: bold
		margin-bottom: 10px
		cursor: pointer
	&__author
		display: flex
	&__number
		font-style: italic
		width: 60px
	&:hover
		background-color: $low-blue-bg

p
	margin: 0
</style>