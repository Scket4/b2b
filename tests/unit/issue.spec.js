import { mount, createLocalVue } from '@vue/test-utils'
import Issue from '@/components/Issue'

const localVue = createLocalVue()
const issue = {
	title: 'Ошибка в асинхронных операциях vue',
	number: '12321',
	created_at: '2012-10-09T23:39:01Z',
	user: {
		login: 'alxxsck',
	},
}

describe('unit tests for issue component', () => {
	it('should init correctly', () => {
		const wrapper = mount(Issue, {
			localVue,
			propsData: { issue },
		})

		expect(wrapper.isVueInstance).toBeTruthy()
		expect(wrapper.is(Issue)).toBeTruthy()
		expect(wrapper.vm.issue).toEqual(issue)
		expect(wrapper.vm.dateCreated).toEqual('09.10.2012')
	})

	it('should render correctly', () => {
		const headerText = `Автор: ${issue.user.login} от 09.10.2012`
		const titleText = issue.title
		const numberText = `# ${issue.number}`
		const wrapper = mount(Issue, {
			localVue,
			propsData: { issue },
		})

		expect(wrapper.find('.issue__author--date').text()).toEqual(headerText)
		expect(wrapper.find('.issue__tag').text()).toEqual(titleText)
		expect(wrapper.find('.issue__number').text()).toEqual(numberText)
	})

	it('should match the snapshot', () => {
		const wrapper = mount(Issue, {
			localVue,
			propsData: { issue },
		})

		expect(wrapper.html()).toMatchSnapshot()
	})
})
