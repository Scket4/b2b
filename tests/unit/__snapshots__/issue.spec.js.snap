// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests for issue component should match the snapshot 1`] = `
<div class="issue">
  <div class="issue__main-info">
    <div class="issue__tag">
      Ошибка в асинхронных операциях vue
    </div>
    <div class="issue__author">
      <p class="issue__author--date">
        Автор: alxxsck от 09.10.2012
      </p>
    </div>
  </div>
  <div class="issue__number"># 12321</div>
</div>
`;
